src/analysis/souffle/taint.dl

```diff
@@ -130,6 +130,7 @@ mayHaveTag(base, tag) :-
   isAccessPath(base),
   isAccessPath(member),
   mayHaveTag(member, tag),
-  contains(cat(base, "."), member).
+  // The member is a member of the base if it starts with the base plus '.'.
+  substr(member, 0, strlen(base) + 1) = cat(base, ".").
```

Comment:
Can we define a datalog relation like `isBaseOf` (or some relevant name) and write some tests?

Reply:
I've done something similar with `isMemberOf`.